libs/wingc/src/type_check.rs

```diff
@@ -268,13 +299,31 @@ impl Subtype for Type {
 				false
 			}
 			(Self::Resource(l0), Self::Resource(_)) => {
-				// If we extend from `other` than I'm a subtype of it (inheritance)
+				// If we extend from `other` then I'm a subtype of it (inheritance)
 				if let Some(parent) = l0.parent.as_ref() {
 					let parent_type: &Type = &*parent;
 					return parent_type.is_subtype_of(other);
 				}
 				false
 			}
+			(Self::Interface(l0), Self::Interface(_)) => {
+				// If we extend from `other` then I'm a subtype of it (inheritance)
```

Comment:
It's handled as a case at the beginning of the function:

```ts
if std::ptr::eq(self, other) {
	return true;
}
```

Reply:
No it isn't, the `std::ptr::eq` compares the pointers so if we're comparing two references to the same Type it'll return `true`. But if we're comparing two types that are equal but are actually two copies in the global `types` array then we won't catch it. This can happen for example with function signatures or baby generics where we create multiple instances of the same type in the global array (because we're lazy).

I guess I'll need to add a test proving this fails and then open an issue..